File: api-chi/cmd/routes/movie_depend_movie_category.go
package routes

import (
	"api-chi/cmd/controllers"

	"github.com/go-chi/chi/v5"
)

func MovieDependMovieCategoryRoutes(r chi.Router) {
	controller := controllers.MovieDependMovieCategoryController{}

	r.Route("/movie-depend-movie-category", func(r chi.Router) {
		r.Get("/count", controller.Count)
		r.Post("/", controller.Create)
		r.Delete("/{id}", controller.Remove)
	})
}

File: api-fastify/src/api/routes/movie_depend_movie_category.ts
import { FastifyInstance } from "fastify"
import MovieDependMovieCategoryController from "@/api/controllers/movie_depend_movie_category"

export default async function MovieDependMovieCategoryRoutes(fastify: FastifyInstance) {
  const controller: MovieDependMovieCategoryController = new MovieDependMovieCategoryController()

  fastify.post("/movie-depend-movie-categories", controller.Create.bind(controller))
  fastify.delete("/movie-depend-movie-categories/:id", controller.Remove.bind(controller))
}
